ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/ShelleyHFC.hs

```diff
@@ -198,16 +207,35 @@ instance ShelleyBasedEra era => HasPartialLedgerConfig (ShelleyBlock era) where
   type PartialLedgerConfig (ShelleyBlock era) = ShelleyPartialLedgerConfig era
 
   -- Replace the dummy 'EpochInfo' with the real one
-  completeLedgerConfig _ epochInfo (ShelleyPartialLedgerConfig cfg _) =
+  completeLedgerConfig _ epochInfo_ (ShelleyPartialLedgerConfig cfg _) =
       cfg {
           shelleyLedgerGlobals = (shelleyLedgerGlobals cfg) {
               SL.epochInfoWithErr =
                   hoistEpochInfo
                     (runExcept . withExceptT (T.pack . show))
-                    epochInfo
+                    epochInfo_
             }
         }
 
+-- | This instance uses the invariant that the EpochInfo in a
+-- ShelleyPartialLedgerConfig is always just a dummy value.
+instance ShelleyBasedEra era
+      => SerialiseNodeToClient (ShelleyBlock era) (ShelleyPartialLedgerConfig era) where
+  decodeNodeToClient ccfg version = do
+    enforceSize "ShelleyPartialLedgerConfig era" 2
+    ShelleyPartialLedgerConfig
+      <$> decodeShelleyLedgerConfigWith (pure (toPureEpochInfo dummyEpochInfo)) ccfg version
+      <*> decodeNodeToClient @(ShelleyBlock era) @TriggerHardFork  ccfg version
+
+  encodeNodeToClient ccfg version
+    (ShelleyPartialLedgerConfig
+      shelleyLedgerConfig
+      triggerHardFork
+    )
+      = encodeListLen 2
+        <> encodeShelleyLedgerConfigWith mempty ccfg version shelleyLedgerConfig
```

Comment:
list length used in `encodeShelleyLedgerConfigWith` will be wrong since we use `mempty` here